libraries/AP_NavEKF3/AP_NavEKF3_MagFusion.cpp

```diff
@@ -264,7 +264,10 @@ void NavEKF3_core::SelectMagFusion()
     magFusePerformed = false;
 
     // Handle case where we are using an external yaw sensor instead of a magnetomer
-    if ((frontend->_magCal == 5)) {
+    MagCal magcal = effective_magCal();
+    if (magcal == MagCal::EXTERNAL_YAW || magcal == MagCal::EXTERNAL_YAW_FALLBACK) {
+        bool have_fused_gps_yaw = false;
+        uint32_t now = AP_HAL::millis();
         if (storedYawAng.recall(yawAngDataDelayed,imuDataDelayed.time_ms)) {
```

Comment:
The problem here is that if there are no yaw measurements received, no yaw fusion of any type will be performed and the filter will eventually diverge. As discussed in our call I will rework it to integrate this selection logic with a movement check when on ground so that yaw fusion always run with either the yaw measurement if available. If not available and no movement is detected, the last predicted yaw recorded when movement was detected will be used. When moving a zero innovation will be used.